coded_tools/brave_search/brave_search.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import os
+from typing import Any, Dict, Union
+
+import requests
+from neuro_san.interfaces.coded_tool import CodedTool
+
+
+class BraveSearch(CodedTool):
+    """
+    CodedTool implementation which provides a way to search the web using Brave search API
+    For info on Brave search, and to get a Brave search API key, go to https://brave.com/search/api/
+    """
+
+    def __init__(self):
+        self.top_n = 5
+        self.brave_api_key = os.getenv("BRAVE_API_KEY")
+        if self.brave_api_key is None:
+            logging.error("BRAVE_API_KEY is not set!")
+        self.brave_url = os.getenv("BRAVE_URL")
```

Comment:
I can see not having an API key as being an error, but shouldn't there be reasonable defaults for URL and TIMEOUT?